test/jsonwp-proxy/proxy-specs.js

```diff
@@ -1,186 +1,195 @@
-      j.command('/badurl', 'GET').should.eventually.be.rejectedWith('Could not proxy');
-    });
-    it('should throw when a command fails', async function () {
-      let j = mockProxy({sessionId: '123'});
-      let e = null;
-      try {
-        await j.command('/element/bad/text', 'GET');
-      } catch (err) {
-        e = err;
-      }
-      should.exist(e);
-      e.message.should.contain('Invisible element');
-    });
-    it('should throw when a command fails with a 200 because the status is not 0', async function () {
-      let j = mockProxy({sessionId: '123'});
-      let e = null;
-      try {
-        await j.command('/element/200/text', 'GET');
-      } catch (err) {
-        e = err;
-      }
-      should.exist(e);
-      e.error.should.eql('element not visible');
-    });
-    it('should throw when a command fails with a 100', async function () {
-      let j = mockProxy({sessionId: '123'});
-      let e = null;
-      try {
-        await j.command('/session/badchrome/nochrome', 'GET');
-      } catch (err) {
-        e = err;
-      }
-      should.exist(e);
-      e.message.should.contain('chrome not reachable');
-    });
-  });
-  describe('req/res proxy', function () {
-    it('should successfully proxy via req and send to res', async function () {
-      let j = mockProxy();
-      let [req, res] = buildReqRes('/status', 'GET');
-      await j.proxyReqRes(req, res);
-      res.headers['content-type'].should.equal('application/json; charset=utf-8');
-      res.sentCode.should.equal(200);
-      res.sentBody.should.eql({status: 0, value: {foo: 'bar'}});
-    });
-    it('should rewrite the inner session id so it doesnt change', async function () {
-      let j = mockProxy({sessionId: '123'});
-      let [req, res] = buildReqRes('/element/200/value', 'GET');
-      await j.proxyReqRes(req, res);
-      res.sentBody.should.eql({status: 0, value: 'foobar', sessionId: '123'});
-    });
-    it('should rewrite the inner session id with sessionId in url', async function () {
-      let j = mockProxy({sessionId: '123'});
-      let [req, res] = buildReqRes('/wd/hub/session/456/element/200/value', 'POST');
-      await j.proxyReqRes(req, res);
-      res.sentBody.should.eql({status: 0, value: 'foobar', sessionId: '456'});
-    });
-    it('should pass through urls that do not require session IDs', async function () {
-      let j = mockProxy({sessionId: '123'});
-      let [req, res] = buildReqRes('/wd/hub/status', 'GET');
-      await j.proxyReqRes(req, res);
-      res.sentBody.should.eql({status: 0, value: {'foo': 'bar'}});
-    });
-    it('should proxy strange responses', async function () {
-      let j = mockProxy({sessionId: '123'});
-      let [req, res] = buildReqRes('/nochrome', 'GET');
-      await j.proxyReqRes(req, res);
-      res.sentCode.should.equal(100);
-      res.sentBody.should.eql({status: 0, value: {message: 'chrome not reachable'}});
-    });
-  });
-});
+// transpile:mocha
```

Comment:
there were a bunch of \r characters at the end of lines in this file for some reason, and when i removed them the whole file got readded to git. not sure why.